modules/effects/spec/effect_sources.spec.ts

```diff
@@ -51,15 +56,52 @@ describe('EffectSources', () => {
         return { type: '[EffectWithInitAction] Init' };
       }
     }
+    const store = TestBed.get(Store);
 
     effectSources.addEffects(new EffectWithInitAction());
 
-    const store = TestBed.get(Store);
+    expect(store.dispatch).toHaveBeenCalledTimes(1);
     expect(store.dispatch).toHaveBeenCalledWith({
       type: '[EffectWithInitAction] Init',
     });
   });
 
+  it('should only dispatch an action on ngrxOnInitEffects once after being registered', () => {
+    class EffectWithInitAction implements OnInitEffects {
+      ngrxOnInitEffects() {
+        return { type: '[EffectWithInitAction] Init' };
+      }
+    }
+    const store = TestBed.get(Store);
+
+    effectSources.addEffects(new EffectWithInitAction());
+    effectSources.addEffects(new EffectWithInitAction());
+
+    expect(store.dispatch).toHaveBeenCalledTimes(1);
+  });
```

Comment:
Can we please add a similar test but where effects class actually has 2 effects? (and should be called once)

Reply:
I addressed the feedback - thanks!